src/lib/ArrayInfo.ts

```diff
@@ -1,121 +1,179 @@
-import {DataError} from "./Errors"
-import {KeyValue} from "./Utils"
+import { DataError } from './Errors'
+import { KeyValue } from './Utils'
 
 function isInt(value: any) {
-    return !isNaN(value) &&
-        Number(value) == value && !isNaN(parseInt(value, 10))
+  return !isNaN(value) && Number(value) == value && !isNaN(parseInt(value, 10))
 }
 
-export const arrayRegex = () => /^([\.0-9a-zA-Z_$\-][0-9a-zA-Z_\-$\.]*)\[((?!(\]|\[)).*|)\]$/gm
+export const arrayRegex = () => /^([\.0-9a-zA-Z_$\-][0-9a-zA-Z_\-$\.]*)(.*)/gm
```

Comment:
Couldn't this still work with the previous regex ?

The previous regex was created with a security researcher to avoid possible [Backtracking in Regular Expressions](https://docs.microsoft.com/en-us/dotnet/standard/base-types/backtracking-in-regular-expressions).

The `(.*)` should be avoided at all cost.

Also, seeing how the regex is used, its old format should still be good to match any type of array (nested or not). After all, the extraction of the indices is done later with the following regex.

Reply:
The existing regex works great with a single index, but not for nested indexes.  I think a simple approach is to do a match on just the first part, and then substring out that match from the full string.

https://regex101.com/
<img width="1178" alt="image" src="https://user-images.githubusercontent.com/93680615/176588969-2567f08f-03e1-4bda-8f7f-e30e3b1b7751.png">

I will build a simple function to extract the indexes out.

Reply:
Regex has been reverted, and function added to support parsing of index values.